### PR TITLE
[DRAFT] Fix mixed target framework issue

### PR DESCRIPTION
**Pull request by mccorkill1**
The swift_dyanmic_framework_aspect was really only meant to run when one of the ios_dynamic_framework, tvos_dynamic_framework, or watchos_dynamic_framework rules are run. This PR fixes this issue https://github.com/bazelbuild/rules_apple/issues/1038 and adds a test to make sure that apps built using mixed target ios_frameworks compile properly.

### CODE REVIEW

File: apple/internal/aspects/swift_dynamic_framework_aspect.bzl [hunk 1]
```diff
@@ -80,6 +82,9 @@ framework module {module_name} {{
 def _swift_dynamic_framework_aspect_impl(target, ctx):
     """Aspect implementation for Swift dynamic framework support."""
 
+    if not ctx.rule.kind == "ios_dynamic_framework" or not ctx.rule.kind == "tvos_dynamic_framework" or not ctx.rule.kind == "watchos_dynamic_framework":
```
Comment (thii) [inline apple/internal/aspects/swift_dynamic_framework_aspect.bzl:85]:
```suggestion
    if ctx.rule.kind not in ["ios_dynamic_framework", "tvos_dynamic_framework", "watchos_dynamic_framework"]:
```

File: test/starlark_tests/targets_under_test/ios/BUILD [hunk 2]
```diff
@@ -1126,6 +1144,24 @@ ios_framework(
...
+        "iphone",
+        "ipad",
+    ],
+    infoplists = [
+        "//test/starlark_tests/resources:Info.plist",
+    ],
+    minimum_os_version = "8.0",
+    tags = FIXTURE_TAGS,
+    deps = [
+        "//test/starlark_tests/resources:swift_main_lib",
```
Comment (thii) [inline test/starlark_tests/targets_under_test/ios/BUILD:1160]: Can you change this to `swift_shared_lib`? This ensures the case when there's more than one swift_library and one objc_library in the transitive deps is verified.

```suggestion
        "//test/starlark_tests/resources:swift_shared_lib",
```

**Comment by mccorkill1** [comment]
I have a different fix coming soon from a different branch I haven't pushed yet. This fix broke some of the tests so I had to approach it a little differently.

**Comment by brentleyjones** [comment]
Thank @mccorkill1. We've now run into this same issue. Eagerly looking forward to your fix.

**Comment by mccorkill1** [comment]
@brentleyjones I just opened the PR https://github.com/bazelbuild/rules_apple/pull/1043

**Pull request closed without merging**